packages/web-react/src/components/ControlButton/demo/ControlButtonDemoFactory.tsx

```diff
@@ -0,0 +1,100 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Intensity, Sizes } from '../../../constants';
+import type {
+  AccentColorNamesType,
+  BackgroundColorsDictionaryType,
+  EmotionColorsDictionaryType,
+  IntensityDictionaryType,
+  NeutralColorsDictionaryType,
+} from '../../../types';
+import { Flex } from '../../Flex';
+import { Icon } from '../../Icon';
+import ControlButton from '../ControlButton';
+
+type ColorType = 'basic' | 'neutral' | 'emotion' | 'accent';
+
+type ControlButtonDemoFactoryProps = {
+  color:
+    | BackgroundColorsDictionaryType
+    | NeutralColorsDictionaryType
+    | EmotionColorsDictionaryType
+    | AccentColorNamesType;
+  colorType: ColorType;
+};
+
+const ControlButtonDemoFactory = ({ color, colorType }: ControlButtonDemoFactoryProps) => {
+  const sizes = Object.values(Sizes);
+
+  const getBackgroundClass = (intensity: IntensityDictionaryType) => {
+    switch (colorType) {
+      case 'basic':
+        return `bg-${color}`;
+      case 'neutral':
+        return `bg-${color}-${intensity}`;
+      case 'emotion':
+        return `bg-emotion-${color}-${intensity}`;
+      case 'accent':
+        return `bg-accent-${color}-${intensity}`;
+      default:
+        return '';
+    }
+  };
+
+  const getTextClass = (intensity: IntensityDictionaryType) => {
+    switch (colorType) {
+      case 'basic':
+        return '';
+      case 'neutral':
+        return `text-${color}-${intensity === Intensity.BASIC ? Intensity.SUBTLE : Intensity.BASIC}`;
```

Comment:
`intensity === Intensity.BASIC ? Intensity.SUBTLE : Intensity.BASIC` → separate to custom function?